Library/Homebrew/cask/lib/hbc/cli/internal_appcast_checkpoint.rb

```diff
@@ -0,0 +1,61 @@
+module Hbc
+  class CLI
+    class InternalAppcastCheckpoint < InternalUseBase
+      def self.run(*args)
+        calculate = args.include? "--calculate"
+        cask_tokens = cask_tokens_from(args)
+        raise CaskUnspecifiedError if cask_tokens.empty?
+
+        if cask_tokens.all? { |t| t =~ %r{^https?://} && t !~ /\.rb$/ }
+          appcask_checkpoint_for_url(cask_tokens)
+        else
+          appcask_checkpoint(cask_tokens, calculate)
+        end
+      end
+
+      def self.appcask_checkpoint_for_url(urls)
```

Comment:
This will work the same both with or without `--calculate`. Just pointing it out in case you intended to do it any other way. It’s fine by me as it is since the only sensical thing to do with a URL is calculate, anyway.